src/libertem/udf/feature_vector_maker.py

```diff
@@ -0,0 +1,90 @@
+from libertem.udf import UDF
+from libertem.masks import _make_circular_mask
+from libertem.udf.stddev import run_stddev
+
+from skimage.feature import peak_local_max
+
+
+class FeatureVecMakerUDF(UDF):
+    def get_result_buffers(self):
+        coordinates = self.params.coordinates
+        return {
+            'feature_vec': self.buffer(
+                kind="nav", extra_shape=(coordinates.shape[0],), dtype="bool"
+            ),
+        }
+
+    def process_frame(self, frame):
+        delta = self.params.delta
+        savg = self.params.savg
+        coordinates = self.params.coordinates
+        ref = savg[coordinates[:, 0], coordinates[:, 1]]
+        for j in range(0, coordinates.shape[0]):
+            if frame[coordinates[j, 0], coordinates[j, 1]]-ref[j] > delta:
+                self.results.feature_vec[j] = 1
+        return
+
+
+def make_feature_vec(ctx, dataset, num, delta, center=None, rad_in=None, rad_out=None, roi=None):
+    """
+    Return a value after integration of Fourier spectrum for each frame over ring.
+    Parameters
+    ----------
+    ctx: Context
+        Context class that contains methods for loading datasets,
+        creating jobs on them and running them
+
+    dataset: DataSet
+        A dataset with 1- or 2-D scan dimensions and 2-D frame dimensions
+
+    num: int
+        Number of possible peak positions to detect (better put higher value,
+         the output is limited to number of peaks the algorithm could find)
+
+    delta: int
+        Intensity difference between current frame and reference image for decision
+        making for feature vector value
+
+    rad_in: int, optional
+        Inner radius in pixels of a ring to mask region of interest of SD image to delete outliers
+        for peak finding
+
+    rad_out: int, optional
+        Outer radius in pixels of a ring to mask region of interest of SD image to delete outliers
+        for peak finding
+
+    center: tuple, optional
+        (y,x) - pixels, coordinates of a ring to mask region of interest of SD image 
```

Comment:
There is some trailing white space here (and on some other lines, see https://travis-ci.org/LiberTEM/LiberTEM/jobs/542256575 for details)